### PR TITLE
Cls17 17 adding test parameters

### PR DESCRIPTION
**Pull request by 09041985**
@dmitrytokarev Take a look please.

### CODE REVIEW

File: tests/test_functons.py [hunk 1]
```diff
@@ -12,56 +12,88 @@
 
 
 @pytest.mark.parametrize("test_square_input,test_square_expected", [
+    (-(3037000499**2-1), -sys.maxsize-1),
+    (-(3037000499**2), -sys.maxsize),
```
Comment (dmitrytokarev) [inline tests/test_functons.py]: actually it was my fault - we don't need extra parentheses here, so you can just
`-3037000499**2-1`
Ok, i think you accidentally switched input/expected output... We partition input values into equivalent partitions. So the boundaries for those partitions are `-sys.maxsize-1`, `0`, `sys.maxsize`
so you have to apply boundary condition analysis to those boundaries (x-1, x, x+1) and pick one values from the each partition: `-sys.maxsize**2`, `-sys.maxsize**0.5` (power of 0.5 == square root), `sys.maxsize**0.5`, and `sys.maxsize**2`.

File: tests/test_functons.py [hunk 2]
```diff
@@ -12,56 +12,88 @@
...
+    (3037000499**2, sys.maxsize)
 ])
 def test_square(test_square_input, test_square_expected):
     assert square(test_square_input) == test_square_expected
 
 
 @pytest.mark.parametrize("test_input,expected", [
-    (-100, -99),
+    (-sys.maxsize,  -sys.maxsize+1),
+    (-sys.maxsize-1,  -sys.maxsize),
```
Comment (dmitrytokarev) [inline tests/test_functons.py]: the boundary here is `-sys.maxsize-1` so need (x-1, x, x+1), you only got 2 inputs for this boundary, missing `x-1` which will be `-sys.maxsize-2`

File: tests/test_functons.py [hunk 3]
```diff
@@ -12,56 +12,88 @@
...
+    (-3037000499, -3037000498),
     (-1, 0),
     (0, 1),
     (1, 2),
-    (100, 101),
+    (3037000499, 3037000500),
+    (sys.maxsize-1, sys.maxsize),
+    (sys.maxsize, sys.maxsize+1)
```
Comment (dmitrytokarev) [inline tests/test_functons.py]: missing `sys.maxsize+1`

File: tests/test_functons.py [hunk 4]
```diff
@@ -12,56 +12,88 @@
...
 ])
 def test_one_good_turn(test_input, expected):
     assert one_good_turn(test_input) == expected
 
 
 @pytest.mark.parametrize("test_square_expected,test_deserves_another_expected", [
-    (-100, -97),
+    (-sys.maxsize,  -sys.maxsize+3),
+    (-sys.maxsize-1,  -sys.maxsize+2),
+    (-3037000499, -3037000496),
```
Comment (dmitrytokarev) [inline tests/test_functons.py]: missing `-sys.maxsize-2`

File: tests/test_functons.py [hunk 5]
```diff
@@ -12,56 +12,88 @@
...
     (-1, 2),
     (0, 3),
     (1, 4),
-    (100, 103),
+    (3037000499, 3037000502),
+    (sys.maxsize-1, sys.maxsize+2),
+    (sys.maxsize, sys.maxsize+3)
```
Comment (dmitrytokarev) [inline tests/test_functons.py]: missing `sys.maxsize+1`

File: tests/test_functons.py [hunk 6]
```diff
@@ -12,56 +12,88 @@
...
 ])
 def test_deserves_another(test_square_expected, test_deserves_another_expected):
     assert deserves_another(test_square_expected) == test_deserves_another_expected
 
 
 def test_biggest_number():
-    assert biggest_number(1,2,3,4,5) == 5
+    assert biggest_number(-sys.maxsize, -sys.maxsize-1, -3037000499, -1, 0, 1, 3037000499,
+                          sys.maxsize-1, sys.maxsize) == 9223372036854775807
```
Comment (dmitrytokarev) [inline tests/test_functons.py]: you should replace `9223372036854775807` with `sys.maxsize`

File: tests/test_functons.py [hunk 7]
```diff
@@ -12,56 +12,88 @@
...
+
 
 
 def test_smallest_number():
-    assert smallest_number(-1,0,1,2,3) == -1
+    assert smallest_number(-sys.maxsize, -sys.maxsize-1, -3037000499, -1, 0, 1, 3037000499,
+                          sys.maxsize-1, sys.maxsize) == -9223372036854775808
```
Comment (dmitrytokarev) [inline tests/test_functons.py]: same here: `-9223372036854775808` >>> `-sys.maxsize`

File: tests/test_functons.py [hunk 8]
```diff
@@ -12,56 +12,88 @@
...
 
 
-def test_distance_from_zeror():
-    assert distance_from_zero(-2) == 2
+@pytest.mark.parametrize("test_distance_from_zero_input,test_distance_from_zero_expected", [
+    (-sys.maxsize,  sys.maxsize),
+    (-sys.maxsize-1,  sys.maxsize+1),
```
Comment (dmitrytokarev) [inline tests/test_functons.py]: missing `-sys.maxsize-2`

File: tests/test_functons.py [hunk 9]
```diff
@@ -12,56 +12,88 @@
...
+    (-3037000499, 3037000499),
+    (-1, 1),
+    (0, 0),
+    (1, 1),
+    (3037000499, 3037000499),
+    (sys.maxsize-1, sys.maxsize-1),
+    (sys.maxsize, sys.maxsize)
```
Comment (dmitrytokarev) [inline tests/test_functons.py]: missing `sys.maxsize+1`

File: tests/test_functons.py [hunk 10]
```diff
@@ -12,56 +12,88 @@
...
+])
+def test_distance_from_zero(test_distance_from_zero_input, test_distance_from_zero_expected):
+    assert distance_from_zero(test_distance_from_zero_input) == test_distance_from_zero_expected
 
 
 @pytest.mark.parametrize("distance_from_zero_type_input,distance_from_zero_type_expected", [
-    (-10, 10),
+    (-sys.maxsize,  sys.maxsize),
+    (-sys.maxsize-1,  sys.maxsize+1),
```
Comment (dmitrytokarev) [inline tests/test_functons.py]: missing `-sys.maxsize-2`

File: tests/test_functons.py [hunk 11]
```diff
@@ -12,56 +12,88 @@
...
+    (-3037000499, 3037000499),
     (-1, 1),
     (0, 0),
+    (1, 1),
+    (3037000499, 3037000499),
+    (sys.maxsize-1, sys.maxsize-1),
+    (sys.maxsize, sys.maxsize),
```
Comment (dmitrytokarev) [inline tests/test_functons.py]: missing `sys.maxsize+1`

File: tests/test_functons.py [hunk 12]
```diff
@@ -12,56 +12,88 @@
...
     ("a", "Nope"),
+    (" ", "Nope")
```
Comment (dmitrytokarev) [inline tests/test_functons.py]: i would add another test: `("", "Nope")` - testing empty string
Also consider other data types: boolean, float, NoneType (None)

File: tests/test_functons.py (hunk 4 above)
Comment (dmitrytokarev) [inline tests/test_functons.py]: also what about `-sys.maxsize**2` - to test a kind of a middle value of the partition `-infinity` to `-sys.maxsize-1`?

File: codecademy/functions.py [hunk 13]
```diff
@@ -1,6 +1,7 @@
 # 4. Function ex-3
 def square(n):
     """This function returns the square of a number."""
+    assert type(n) == int or type(n) == long, "{} was not int".format()
```
Comment (dmitrytokarev) [inline codecademy/functions.py]: So this line didn't cause issues with Python3 accidentally, since all the calls had argument `int` and second part of assertion statement didn't get evaluated (`or type(n) == long`), if you'd call square with non int type it would raise exception in Python3 since they got rid of long type in python3 - it's all int now.
better do this
```python
# add to the end of the import section of this file
import sys
if sys.version_info > (3,):
    long = int
```
then optionally improve your assert statement with
```python
assert isinstance(n, (int, long)),  "{} was not int or long".format()
```

File: codecademy/functions.py [hunk 14]
```diff
@@ -36,7 +37,7 @@ def distance_from_zero(arg):
 # 4. Function ex-19
 def distance_from_zero_type(arg):
     """Depending on the type of argument returned absolute value or 'Nope'"""
-    if type(arg) == int or type(arg) == float:
+    if type(arg) == int or type(arg) == float or type(arg) == long:
```
Comment (dmitrytokarev) [inline codecademy/functions.py]: same thing here you can improve checking the type of the variable by using `isinstance()`

File: tests/test_conditions_and_control_flow.py [hunk 15]
```diff
@@ -9,17 +9,18 @@
...
+    (-sys.maxsize**0.5, -1),
     (-1, -1),
     (0, -1),
     (1, -1),
+    (3, -1),
     (4, -1),
     (5, 0),
     (6, 1),
-    (3037000499, 1),
+    (sys.maxsize**0.5, 1),
```
Comment (dmitrytokarev) [inline tests/test_conditions_and_control_flow.py]: :+1:

**Comment by dmitrytokarev** [comment]
Great job addressing all the comments!
Shipit! :shipit:
